spark/sql-30/src/itest/scala/org/elasticsearch/spark/sql/streaming/api/java/JavaStreamingQueryTestHarness.java

```diff
@@ -0,0 +1,93 @@
+package org.elasticsearch.spark.sql.streaming.api.java;
+
+import java.io.Serializable;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Encoder;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.streaming.DataStreamWriter;
+import org.apache.spark.sql.streaming.StreamingQuery;
+import org.elasticsearch.spark.sql.streaming.StreamingQueryTestHarness;
+import scala.Function0;
+import scala.runtime.AbstractFunction0;
+
+/**
+ * Java based wrapper around the scala base streaming query test harness
+ */
+public class JavaStreamingQueryTestHarness<S extends Serializable> {
+
+    private final StreamingQueryTestHarness<S> harness;
+
+    public JavaStreamingQueryTestHarness(SparkSession spark, Encoder<S> encoder) {
+        this.harness = new StreamingQueryTestHarness<>(spark, encoder);
+    }
+
+    public Dataset<S> stream() {
+        return harness.stream();
+    }
+
+    public JavaStreamingQueryTestHarness<S> withInput(S data) {
+        harness.withInput(data);
+        return this;
+    }
+
+    public JavaStreamingQueryTestHarness<S> setTestTimeout(long timeout, TimeUnit unit) {
+        harness.setTestTimeout(timeout, unit);
+        return this;
+    }
+
+    public JavaStreamingQueryTestHarness<S> expectingToThrow(Class<? extends Throwable> clazz) {
+        harness.expectingToThrow(clazz);
+        return this;
+    }
+
+    public JavaStreamingQueryTestHarness<S> expectingToThrow(Class<? extends Throwable> clazz, String message) {
+        harness.expectingToThrow(clazz, message);
+        return this;
+    }
+
+    public StreamingQuery start(final DataStreamWriter<?> writer, final String path) {
+        Function0<StreamingQuery> runFunction = new AbstractFunction0<StreamingQuery>() {
+            @Override
+            public StreamingQuery apply() {
+                return writer.start(path);
+            }
+        };
+        return harness.startTest(runFunction);
+    }
+
+    public StreamingQuery start(final DataStreamWriter<?> writer) {
+        Function0<StreamingQuery> runFunction = new AbstractFunction0<StreamingQuery>() {
+            @Override
+            public StreamingQuery apply() {
+                return writer.start();
+            }
+        };
+        return harness.startTest(runFunction);
+    }
+
+    public void run(final DataStreamWriter<?> writer, final String path) {
+        Function0<StreamingQuery> runFunction = new AbstractFunction0<StreamingQuery>() {
+            @Override
+            public StreamingQuery apply() {
+                return writer.start(path);
+            }
+        };
+        harness.runTest(runFunction);
+    }
+
+    public void run(final DataStreamWriter<?> writer) {
+        Function0<StreamingQuery> runFunction = new AbstractFunction0<StreamingQuery>() {
+            @Override
+            public StreamingQuery apply() {
+                return writer.start();
```

Comment:
Don't you get a compile time exception here? writer.start
Unhandled exception: java.util.concurrent.TimeoutException

Reply:
no
`./gradlew clean test distZip` seems to be working just fine

environment details:
```
$ cat /etc/*release*
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=20.04
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Ubuntu 20.04 LTS"
NAME="Ubuntu"
VERSION="20.04 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal

$ echo $JAVA_HOME
/usr/lib/jvm/java-8-openjdk-amd64

$ echo $JAVA8_HOME
/usr/lib/jvm/java-8-openjdk-amd64

$ java -version
openjdk version "1.8.0_252"
OpenJDK Runtime Environment (build 1.8.0_252-8u252-b09-1ubuntu1-b09)
OpenJDK 64-Bit Server VM (build 25.252-b09, mixed mode)
```

Reply:
But if you don't set the distro, it will fall in the default case which is 
default:
    String version = project.hadoop22Version	            

I'm building with 
/gradlew -Ddistro=hadoopYarn3

Reply:
```
./gradlew -Pdistro=hadoopStable
./gradlew -Pdistro=hadoopYarn
```
both are now passing
now hadoopYarn3 is still tricky